scripts/detect_headless.js

```diff
@@ -171,30 +171,66 @@ function testImage(resultBlock) {
   body.appendChild(image);
 
   image.onerror = function(){
-    imageWriteResult(resultBlock, image.width, image.height);
+    writeToBlock(resultBlock, `Broken image has width ${image.width} and height ${image.height}`);
     if(image.width === 0 && image.height === 0)
       return HEADLESS;
     return HEADFUL;
   }
 }
 
-function imageWriteResult(resultBlock, width, height) {
-  resultBlock.innerHTML = `Broken image has width ${width} and height ${height}`;
+// Test for outerHeight and outerWidth
+function testOuter(resultBlock) {
+  let outerHeight = window.outerHeight;
+  let outerWidth  = window.outerWidth;
+
+  writeToBlock(resultBlock, `Outerheight: ${outerHeight} and outerwidth: ${outerWidth}`);
+  return (outerHeight === 0 && outerWidth === 0) ? HEADLESS : HEADFUL;
+}
+
+// Test for screenY
+function testScreenY(resultBlock) {
+  let screenY = window.screenY;
+
+  writeToBlock(resultBlock, `ScreenY: ${screenY}`);
+  return screenY === 0 ? HEADLESS : HEADFUL;
+}
+
+// Test for connection-rtt
+function testConnecionRtt(resultBlock) {
+  let connectionRtt = navigator.connection.rtt;
```

Comment:
If `navigator` does not have `connection` attribute it calling `navigator.connection.rtt` throws an exception.
Check if `navigator.connection` isn't `undefined` before calling `.rtt`